frappe/core/doctype/file/file.py

```diff
@@ -445,6 +445,9 @@ def get_full_path(self):
 
 		file_path = self.file_url or self.file_name
 
+		if file_path.startswith(frappe.utils.get_url()) and "/files/" in file_path:
```

Comment:
We should strip site url from files on insert so that this doesn't happen to begin with? 🤔 We can get rid of some dead File records too if it were for this :crie:

---

Also, btw

```suggestion
		if "/files/" in file_path and file_path.startswith(frappe.utils.get_url()):
```

This is better for perf - but I'm not sure we should be doing multiple `frappe.utils.get_url` (which can be costly) calls to begin with.